abcclassroom/clone.py

```diff
@@ -60,21 +87,28 @@ def clone_student_repos(args):
     clone_dir = cf.get_config_option(config, "clone_dir", True)
     organization = cf.get_config_option(config, "organization", True)
     materials_dir = cf.get_config_option(config, "course_materials", False)
+    print(config)
 
     if materials_dir is None:
         print(
-            "No course_materials directory set in config.yml. Will clone "
-            "repositories but will not copy any assignment files."
+            "Oops! I couldn't find a course_materials directory location "
+            "in your config.yml file. I will just clone all of the student"
+            "repositories. I can't copy any assignment files to a "
+            "course_materials directory."
         )
+    # This is a really large try block and makes it hard to figure out where
+    # things fail. perhaps break out into a few sub try / excepts?
     try:
         # Create the assignment subdirectory path and ensure it exists
         Path(course_dir, clone_dir, assignment_name).mkdir(exist_ok=True)
         missing = []
+        # If it can't find the roster file you'll get a FileNotFoundError
+        # Also what happens if the roster is not in the correct format??
         with open(roster_filename, newline="") as csvfile:
             reader = csv.DictReader(csvfile)
             for row in reader:
```

Comment:
check that there is a header row and the columns that we expect .
there is syntax to grab the header and the first row that we can do.. then move on to the for loop??